### ISSUE TITLE
Provider declaration missing in index.d.ts

### ISSUE THREAD

**Issue by ngnathan** [envato/react-breakpoints#11]
Using React with TypeScript, I can't seem to import Provider.  

Looking at the index.d.ts file, it looks like the interface to this is missing.

**Comment by gwitteveen** [comment]
Same goes for the re-export of useResizeObserver.
Basically the exports from `@envato/react-resize-observer-hook` lack type definitions.

**Comment by mdingena** [comment]
I feared this would happen. I'm not too well versed in TS but I added it as a courtesy to the many developers out there who love TS. 

**I could use your help** with adding interfaces to `@envato/react-resize-observer-hook`. I've been working on a branch to add types but it's been on hold because I'm unsure whether I need to manually implement an interface for `ResizeObserver` or not, now that Safari has implemented it and the TypeScript PR seemed to re-opened.

Please see #8 and https://github.com/microsoft/TypeScript/issues/37861 for context.

I will push up a branch with my stale WIP shortly.

**Comment by mdingena** [comment]
@ngnathan @gwitteveen I'm inviting you to contribute to envato/react-resize-observer-hook#3

**Comment by artmsilva** [comment]
Any updates on this? I noticed this issue while trying to build using Rollup.js.

**Comment by mdingena** [comment]
@artmsilva I'm not sure what the correct way forward is. On one hand I'd like to help TS users use this package (I'm not a TS user myself). On the other hand it feels redundant to write the entire `ResizeObserver` interface when [official TypeScript support is being worked on](https://github.com/microsoft/TypeScript/issues/37861).

I suppose I could export it with an explicit `any` to at least suppress errors in consuming packages, then adopt the official interface when it's released later.

What do you think?

**Comment by artmsilva** [comment]
@mdingena I think the latter is best in the interim. Thanks for being so responsive!

**Comment by mdingena** [comment]
I've marked envato/react-resize-observer-hook#3 as **Ready For Review**. It rewrites the JavaScript to TypeScript rather than just providing an `index.d.ts` file.

Once that PR is merged, we can update this package's dependencies to resolve the issue.

@artmsilva @ngnathan @gwitteveen I invite you all to please have a look at the linked PR. There is a lot going but could you please focus on the `.ts` / `.tsx` files and the type definitions therein?

**Comment by mdingena** [comment]
I'm waiting for TypeScript 4.2 to be released, which includes interfaces for `ResizeObserver`.

It's currently scheduled for 23 February: https://github.com/microsoft/TypeScript/issues/41601
